### ISSUE TITLE
Bug: Coarsen operator for density in class INSVCStaggeredNonConservativeHierarchyIntegrator

### ISSUE THREAD

**Issue by nishantnangia** [IBAMR/IBAMR#865]
Hi folks,

The coarsen operator is incorrectly specified in INSVCStaggeredNonConservativeHierarchyIntegrator. Specifically, this line:
https://github.com/IBAMR/IBAMR/blob/master/src/navier_stokes/INSVCStaggeredNonConservativeHierarchyIntegrator.cpp#L989

should use `d_rho_coarsen_type` instead.

Without this fix, all non-conservative VC cases will crash during the initial time step with error message (e.g. examples/vc_navier_stokes/ex0):

> P=00000:Coarsen operator 'CONSERVATIVE_LINEAR_REFINE' not found for variable 'rho_var

The bug originated from this commit:
https://github.com/IBAMR/IBAMR/commit/0b6e65552aeaa7f7f26cbba30c8c095c6aff3fa8#diff-ed1029e41f9b13ac6a438e15afe1d5fbR958
although at a glance, it does not appear that this mistake was made anywhere else.

==
Aside: @boyceg, could you add me back to the IBAMR organization? I have been looking into some multiphase cases and may want to push some branches in the near future.

**Comment by drwells** [comment]
Nice catch. Could you post the input file you used? It would be good to reproduce this and add a test.

**Comment by nishantnangia** [comment]
The input file was input3d from examples/vc_navier_stokes/ex0, but any of
the input files within should exhibit the same behavior — including the 2D
ones.

**Comment by nishantnangia** [comment]
Are there any instructions on how to create new tests (the last time I wrote a test for IBAMR was back when we were using gtest)? If so, could you point me to them -- I'd be happy to take care of this PR.

**Comment by Raamkrishnan** [comment]
Hi, I recently followed this link to write a test:

https://github.com/IBAMR/IBAMR/wiki/Tests-for-IBAMR

**Comment by nishantnangia** [comment]
Ah wonderful, thanks! I'll take a look.

**Comment by amneetb** [comment]
@nishantnangia 

I built `numdiff` on OSX by using:

```
asbhalla@Amneets-MBP-2:~/Softwares/tars/numdiff-5.9.0$ head config.log 
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by Numdiff configure 5.8, which was
generated by GNU Autoconf 2.63.  Invocation command line was

  $ ./configure CC=clang --prefix=/Users/asbhalla/Softwares/NumDiff CFLAGS=-I/usr/local/opt/gettext/include LDFLAGS=-L/usr/local/opt/gettext/lib

```

and IBAMR using `attest` by

asbhalla@Amneets-MBP-2:~/Softwares/IBAMR-Git/objs-dbg$ head config.log 
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by IBAMR configure 171104, which was
generated by GNU Autoconf 2.69.  Invocation command line was

```

  $ ../IBAMR/configure MPIEXEC=/usr/local/bin/mpiexec NUMDIFF=/Users/asbhalla/Softwares/NumDiff/bin/numdiff CFLAGS=-g -O0 -Wall CXXFLAGS=-g -O0 -Wall -std=gnu++11 FCFLAGS=-g -O0 -Wall CC=mpicc CXX=mpicxx FC=mpif90 --with-samrai=/Users/asbhalla/Softwares/SAMRAI/darwin-dbg --with-hdf5=/Users/asbhalla/Softwares/HDF5 --with-silo=/Users/asbhalla/Softwares/SILO --with-boost=/Users/asbhalla/Softwares/boost/1.67.0 --enable-libmesh --with-libmesh=/Users/asbhalla/Softwares/libMesh-Git/darwin-dbg --with-libmesh-method=dbg --enable-gsl --with-gsl=/Users/asbhalla/Softwares/GSL PETSC_DIR=/Users/asbhalla/Softwares/PETSc-BitBucket/PETSc PETSC_ARCH=darwin-dbg --no-create --no-recursion

```

**Comment by drwells** [comment]
Writing a test should be pretty easy for this case - we just need to pare down an example into something that runs in a few seconds and demonstrates the bug (or shows that the bug is fixed). Most of the tests are more-or-less copied and pasted example programs.

Let me know if you have questions about writing a test - its supposed to be easy. If it isn't I would like to know what we can change to make it easier.

**Comment by drwells** [comment]
Fixed by #870.
